src/geometry/query_iterator.cpp

```diff
@@ -60,7 +60,7 @@ IteratorStatus CPPQueryIterator::skip_to(t_docId docId) {
                                            std::ranges::end(iter_), docId);
   index_ = std::ranges::distance(std::ranges::begin(iter_), it + 1);
   if (!has_next()) {
-    return ITERATOR_EOF;
+    base_.atEOF = true;
```

Comment:
but it should also return EOF?

Reply:
No. This was and should be a preparation for the next iteration. We can also remove it completely